metadata-io/src/main/java/com/linkedin/metadata/entity/DeleteEntityService.java

```diff
@@ -0,0 +1,254 @@
+    try {
+      TimeUnit.SECONDS.sleep(seconds);
+    } catch (InterruptedException e) {
+      log.error("Interrupted sleep", e);
+    }
+  }
+
+  /**
+   * Processes an aspect of a given {@link RelatedEntity} instance that references a given {@link Urn}, removes said
+   * urn from the aspects and submits an MCP with the updated aspects.
+   * @param urn           The urn to be found.
+   * @param relatedEntity The entity to be modified.
+   */
+  private void deleteReference(final Urn urn, final RelatedEntity relatedEntity) {
+    final Urn relatedUrn = UrnUtils.getUrn(relatedEntity.getUrn());
+    final String relationType = relatedEntity.getRelationshipType();
+    final String relatedEntityName = relatedUrn.getEntityType();
+
+    log.debug("Processing related entity {} with relationship {}", relatedEntityName, relationType);
+
+    final EntitySpec relatedEntitySpec = _entityService.getEntityRegistry().getEntitySpec(relatedEntityName);
+
+    final java.util.Optional<Pair<EnvelopedAspect, AspectSpec>> optionalAspectInfo =
+        getAspectReferringTo(urn, relatedUrn, relationType, relatedEntitySpec);
+
+    if (!optionalAspectInfo.isPresent()) {
+      log.error("Unable to find aspect information that relates {} {} via relationship {}", urn, relatedUrn, relationType);
+      return;
+    }
+
+    final String aspectName = optionalAspectInfo.get().getKey().getName();
+    final Aspect aspect = optionalAspectInfo.get().getKey().getValue();
+    final AspectSpec aspectSpec = optionalAspectInfo.get().getValue();
+    final List<RelationshipFieldSpec> optionalRelationshipFieldSpec =
+        aspectSpec.findRelationshipFor(relationType, urn.getEntityType());
+
+    if (optionalRelationshipFieldSpec.isEmpty()) {
+      log.error("Unable to find relationship spec information in {} that connects to {} via {}",
+          aspectName, urn.getEntityType(), relationType);
+      return;
+    }
+
+    for (RelationshipFieldSpec relationshipFieldSpec : optionalRelationshipFieldSpec) {
+      final PathSpec path = relationshipFieldSpec.getPath();
+      final Aspect updatedAspect = DeleteReferencesService.getAspectWithReferenceRemoved(urn.toString(), aspect,
+          aspectSpec.getPegasusSchema(), path);
+
+      final MetadataChangeProposal gmce = new MetadataChangeProposal();
+      gmce.setEntityUrn(relatedUrn);
+      gmce.setChangeType(ChangeType.UPSERT);
+      gmce.setEntityType(relatedEntityName);
+      gmce.setAspectName(aspectName);
+      gmce.setAspect(GenericRecordUtils.serializeAspect(updatedAspect));
+
+      final AuditStamp auditStamp = new AuditStamp().setActor(UrnUtils.getUrn(Constants.SYSTEM_ACTOR)).setTime(System.currentTimeMillis());
+      final EntityService.IngestProposalResult ingestProposalResult = _entityService.ingestProposal(gmce, auditStamp);
+
+      if (!ingestProposalResult.isDidUpdate()) {
+        log.warn("Aspect update did not update metadata graph. Before {}, after: {}", aspect, updatedAspect);
```

Comment:
What does this mean? How is this actionable?

Reply:
It means that the MCP did not generate an update in the metadata graph. I've already added a check to only issue MCPs if changes on the aspect were made which means this log is an error state. The operator should check the MCP processor to understand why the MCP did not update (possibly not accepted?)

Reply:
let's log it load and clear with a .error(). 

"Failed to ingest aspect with references removed. Before {} ......"